src/main/java/jenkins/plugins/horreum/BaseExecutionContext.java

```diff
@@ -140,9 +145,18 @@ private static ByteArrayOutputStream toByteArrayOutputStream(InputStream stream)
 
    protected HorreumClient createClient() {
       HorreumClient.Builder clientBuilder = new HorreumClient.Builder()
-            .horreumUrl(url)
-            .horreumUser(usernameCredentials.getUsername())
-            .horreumPassword(usernameCredentials.getPassword().getPlainText());
+            .horreumUrl(url);
+      switch (authenticationType) {
+         case BASIC:
+         case OIDC:
+            clientBuilder.horreumUser(usernameCredentials.getUsername())
+                         .horreumPassword(usernameCredentials.getPassword().getPlainText());
+            break;
+         case API_KEY:
+            clientBuilder.horreumApiKey(keyCredentials.getSecret().getPlainText());
```

Comment:
Theoretically `keyCredentials` could be null.. consider the scenario where you set `API_KEY` but you provide a `usernameAndPassword` credential.